### ISSUE TITLE
Cannot catch exception

### ISSUE THREAD

**Issue by kszafran** [eiiches/jackson-jq#49]
I have encountered a case (I'd think a pretty common one) where I cannot catch an exception with the try-catch expression:

jq:
```
$ jq -n '[82, "WAT", 22] | map(try tonumber)'
[
  82,
  22
]
```

jackson-jq:
```
Exception in thread "main" java.lang.NumberFormatException: For input string: "WAT"
	at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2043)
	at sun.misc.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
	at java.lang.Double.parseDouble(Double.java:538)
	at net.thisptr.jackson.jq.internal.functions.ToNumberFunction.apply(ToNumberFunction.java:22)
	at net.thisptr.jackson.jq.internal.tree.FunctionCall.apply(FunctionCall.java:26)
	at net.thisptr.jackson.jq.internal.tree.TryCatch.apply(TryCatch.java:29)
	at net.thisptr.jackson.jq.internal.FixedScopeQuery.apply(FixedScopeQuery.java:22)
	at net.thisptr.jackson.jq.internal.JsonQueryFunction.applyRecursive(JsonQueryFunction.java:42)
	at net.thisptr.jackson.jq.internal.JsonQueryFunction.apply(JsonQueryFunction.java:36)
	at net.thisptr.jackson.jq.internal.tree.FunctionCall.apply(FunctionCall.java:26)
	at net.thisptr.jackson.jq.internal.tree.PipedQuery.applyRecursive(PipedQuery.java:42)
	at net.thisptr.jackson.jq.internal.tree.PipedQuery.applyRecursive(PipedQuery.java:55)
	at net.thisptr.jackson.jq.internal.tree.PipedQuery.apply(PipedQuery.java:25)
	at net.thisptr.jackson.jq.internal.tree.ArrayConstruction.apply(ArrayConstruction.java:29)
	at net.thisptr.jackson.jq.internal.IsolatedScopeQuery.apply(IsolatedScopeQuery.java:21)
	at net.thisptr.jackson.jq.internal.JsonQueryFunction.applyRecursive(JsonQueryFunction.java:42)
	at net.thisptr.jackson.jq.internal.JsonQueryFunction.applyRecursive(JsonQueryFunction.java:53)
	at net.thisptr.jackson.jq.internal.JsonQueryFunction.apply(JsonQueryFunction.java:36)
	at net.thisptr.jackson.jq.internal.tree.FunctionCall.apply(FunctionCall.java:26)
	at net.thisptr.jackson.jq.internal.tree.PipedQuery.applyRecursive(PipedQuery.java:42)
	at net.thisptr.jackson.jq.internal.tree.PipedQuery.applyRecursive(PipedQuery.java:55)
	at net.thisptr.jackson.jq.internal.tree.PipedQuery.apply(PipedQuery.java:25)
	at net.thisptr.jackson.jq.internal.IsolatedScopeQuery.apply(IsolatedScopeQuery.java:21)
...
```

Seems that `TryCatch` is expecting only a specific type of exceptions:
```
try {
	return tryExpr.apply(scope, in);
} catch (JsonQueryException e) {
	...
}
```
but `ToNumberFunction` is throwing simply a `NumberFormatException`.

**Comment by eiiches** [comment]
Good catch! Thanks for reporting.

This has been fixed in branch/1.x (commit https://github.com/eiiches/jackson-jq/commit/e27c64f71e726c39d26eaf14a2491ea1703753e5) but never been backported to 0.x versions as it should be. I'll fix this.

**Comment by eiiches** [comment]
I'll release v0.0.11 in a few days.

**Comment by kszafran** [comment]
Thanks!

**Comment by eiiches** [comment]
Just released v0.0.11.
